finagle-core/src/test/scala/com/twitter/finagle/netty3/Netty3TransporterTest.scala

```diff
@@ -236,6 +236,29 @@ class Netty3TransporterTest extends FunSpec with MockitoSugar {
 
         assert(!channel.isOpen)
       }
+
+      describe ("handshake timeout") {
```

Comment:
not necessary but if you're feeling ambitious... you could write a test case covering the symptom you describe in https://github.com/twitter/finagle/issues/345, specifically, a hung ssl handshake causing the channel to get stuck in the connecting state and showing how this new timeout closes the channel.